compliance-monitor/monitor.py

```diff
@@ -421,36 +430,42 @@ async def post_report(
     sep += len(SEP)
     signature = body_text[:sep - 1]  # do away with the ampersand!
     body_text = body_text[sep:]
-    json_text = None
+    try:
+        ssh_validate(keys, signature, body_text)
+    except ValueError:
+        raise HTTPException(status_code=401)
```

Comment:
Potential problem: same status code used as above and no specific error message in the response, which can make debugging difficult for clients using the endpoint. (Of course, we also should not reveal too much information.)

Reply:
Yes, I didn't want to reveal too much, but I could add a misty nebulous message

Reply:
772b60390bfa214742e1b120fc5bf1a266bc2c41